### PR TITLE
refactor: Tree in `Flyout` and `InlineDialog` support

### PR DESCRIPTION
**Pull request by Kenny806**
This ensures that Tree in compatible with the new `InlineDialog`.

### CODE REVIEW

**Review by anxobotana — APPROVED** [review]
LGTM, but it seems that we need to revisit how the background behaves in a flyout with draggable items.

File: src/components/Tree/Tree.stories.tsx [hunk 1]
```diff
@@ -327,3 +328,20 @@ export const WithExpandOnSelect = ({ ...args }: TreeProps) => {
...
     );
 };
+
+export const InsideFlyout = ({ ...args }: TreeProps) => {
+    const [isOpen, setIsOpen] = useState(false);
+
+    return (
+        <div style={{ width: 800 }}>
+            <Flyout
+                trigger={<button type={'button'}>{'Open Tree'}</button>}
```
Comment (imoutaharik) [inline src/components/Tree/Tree.stories.tsx:338]: It's fine for me to add this story but can we style a bit so it doesn't look like bare minimum. I would use a Fondue Button here. 
`trigger={<Button>Open Tree</Button>}`

File: src/components/Tree/Tree.stories.tsx [hunk 2]
```diff
@@ -327,3 +328,20 @@ export const WithExpandOnSelect = ({ ...args }: TreeProps) => {
         </div>
     );
 };
+
+export const InsideFlyout = ({ ...args }: TreeProps) => {
+    const [isOpen, setIsOpen] = useState(false);
+
+    return (
+        <div style={{ width: 800 }}>
```
Comment (imoutaharik) [inline src/components/Tree/Tree.stories.tsx:336]: Just a detail here but you could use our Layout component instead of div -> either `Box` or `Container`

**Comment by anxobotana** [comment]
@imoutaharik I will take the PR and fix among those things you commented, the background issues when in a flyout.

File: src/components/Tree/Tree.stories.tsx [hunk 3]
```diff
@@ -327,3 +333,50 @@ export const WithExpandOnSelect = ({ ...args }: TreeProps) => {
...
+                    <DialogBody>
+                        <TreeView
+                            {...args}
+                            expandedIds={expandedIds}
+                            onExpand={handleItemExpand}
+                            onShrink={handleItemShrink}
+                        >
+                            {treeItemsMock.map(renderTreeItemLabel)}
+                        </TreeView>
+                    </DialogBody>{' '}
```
Comment (imoutaharik) [inline src/components/Tree/Tree.stories.tsx:377]: what's the {'  '} for?

Comment (anxobotana) [reply src/components/Tree/Tree.stories.tsx]: nothing... just missed it when cleaning :)  good catch!
cleaned now!

**Review by imoutaharik — APPROVED** [review]

**Pull request merged**